### PR TITLE
Bugfix For handover App

### PR DESCRIPTION
**Pull request by vinay-ebi**
1) Fix for handover status never update on dropdatabase fail
2) Fix for Compara dispatch trigger error UI
3) Multiple selection to delete handover token

### CODE REVIEW

**Review by marcoooo — CHANGES_REQUESTED** [review]
Please could you add corresponding unit test, if possible?

**Comment by vinay-ebi** [comment]
> Please could you add corresponding unit test, if possible?

It require to setup mysql,  i will make sure this test case covers in up coming PR

File: src/ensembl/production/handover/celery_app/utils.py [hunk 1]
```diff
@@ -109,23 +109,31 @@ def get_tgt_uri(src_url, staging_uri):
     return '%s%s' % (staging_uri, src_url.database)
 
 
-def drop_current_databases(current_db_list, spec):
+def drop_current_databases(current_db_list, spec, target_db_delete=None):
     """Drop databases on a previous assembly or previous genebuild (e.g: Wormbase) from the staging MySQL server"""
```
Comment (marcoooo) [inline src/ensembl/production/handover/celery_app/utils.py:113]: Add parameters description.

File: src/ensembl/production/handover/celery_app/utils.py [hunk 2]
```diff
@@ -109,23 +109,31 @@ def get_tgt_uri(src_url, staging_uri):
...
-    else:
-        for database in current_db_list:
-            db_uri = staging_uri + database
-            if database_exists(db_uri):
-                msg = 'Dropping %s' % db_uri
-                log_and_publish(make_report('INFO', msg, spec, tgt_uri))
-                drop_database(db_uri)
+    try:
+        
+        if target_db_delete:
```
Comment (marcoooo) [inline src/ensembl/production/handover/celery_app/utils.py:116]: I am not sure what this give us here, can you explain it to me please? the initial point of this method is to delete all related dbs, adding the parameter is to force something, but I don't  understand.

**Review by marcoooo — CHANGES_REQUESTED** [review]

File: src/ensembl/production/handover/celery_app/utils.py (hunk 2 above)
Comment (vinay-ebi) [reply src/ensembl/production/handover/celery_app/utils.py:116]: Initally drop_database(spec['tgt_uri']) fails to change log message as celery task is not under try except . I tried to use existing  function drop_current_databases in utils.py with new param target_db_delete to drop given db src, as it contains another functionality to drop db based on previous assembly or previous genebuild.
if dropdatabase fails to drop the given db it raise the exception and  return false so log message is reported to es and relected in handover UI

File: src/ensembl/production/handover/celery_app/utils.py [hunk 3]
```diff
@@ -109,23 +109,41 @@ def get_tgt_uri(src_url, staging_uri):
...
-                msg = 'Dropping %s' % db_uri
-                log_and_publish(make_report('INFO', msg, spec, tgt_uri))
-                drop_database(db_uri)
+def drop_current_databases(current_db_list, spec, target_db_delete=None):
+    """[Drop databases on a previous assembly or previous genebuild (e.g: Wormbase) from the staging MySQL server]
+
+    Args:
+        current_db_list ([List]): [List of databases names to be dropped]
+        spec ([Dict]): [Handover payload with details ]
+        target_db_delete ([Boolean], optional): [Flag to delete givem list of databases]. Defaults to None.
```
Comment (marcoooo) [inline src/ensembl/production/handover/celery_app/utils.py:118]:
```suggestion
        target_db_delete ([Boolean], optional): [Flag to delete given list of databases]. Defaults to None.
```

**Review by marcoooo — APPROVED** [review]

**Pull request merged**